src/lib/service/handlers.ts

```diff
@@ -0,0 +1,108 @@
+import { CreateContext, DeleteContext, DetailContext, ListContext, UpdateContext } from 'lib/context/crudContext';
+import { Operation } from 'lib/context/operation';
+import { errorOrEmpty } from 'lib/helpers';
+import { ServiceImplementation } from 'lib/settings/definitions';
+
+export const createHandlers = <T extends { id: any }, C extends object>(
+    implementation: ServiceImplementation<T, C>
+) => {
+    /**
+     * Fetch resource, throw error when resource missing.
+     * This method is used for handlers working with a single existing resource (get, update, delete)
+     */
+    const getSafe = (context: Pick<DetailContext<T, C>, 'id' | 'context' | 'options'>): Promise<T> =>
+        implementation
+            .detail({ ...context, type: Operation.DETAIL, write: false, safe: true })
+            .then(errorOrEmpty(implementation.createNotFoundError()));
+
+    const detailHandler = (options: any = {}) => async (id: number, context: C) => {
+        const dynamicOptions = await implementation.getOptions(Operation.DETAIL);
+        options = { ...dynamicOptions, ...options, ...(context as object) };
+        const ctx: DetailContext<T, C> = await getSafe({ id, context, options }).then(entity => ({
```

Comment:
Please keep the same formatting of `.then` callings accross the project.

- `}).then(` on the same line here
- `).then(` on a newline on l.38 ...

Reply:
It is consistent and automatically managed by prettier. If you dont like that, please suggest changes in https://github.com/AckeeCZ/prettier-config-ackee.

Notice that the blocks are not identical (even ignoreing line length, which generally are taken into account) -- this line is preceeded by `}` while 38 by `)`